app/models/policy.rb

```diff
@@ -8,6 +8,16 @@ class Policy < ApplicationRecord
   validate :belongs_to_a_tenant
   validate :validate_schema_specification
 
+  def self.find_by_id_or_name_version(id_or_name_version)
```

Comment:
This look like a slug. You can reuse `has_permalink` unique per account

Reply:
permalink replaces dots with dashes, making the identifier to loose its purpose a little bit. The customer would have to know, for example, that for a policy whose name is "my-policy" and version is "1.0", the identifier would be "my-policy-1-0" instead of "my-policy-1.0".